bin/install

```diff
@@ -1,105 +1,149 @@
-      export MACOSX_DEPLOYMENT_TARGET=$(/usr/bin/sw_vers --productVersion)
+    if [[ "$OSTYPE" = "darwin"* ]] && [[ -z "${MACOSX_DEPLOYMENT_TARGET-}" ]]; then
+      MACOSX_DEPLOYMENT_TARGET=$(/usr/bin/sw_vers --productVersion)
+      export MACOSX_DEPLOYMENT_TARGET
     fi
 
-    echo "Building luaJIT with options: $luaJIT_configure_options"
-    make amalg $luaJIT_configure_options || exit 1
-    make install $luaJIT_configure_options || exit 1
+    printf "Building luaJIT with options:"
+    printf " %q" "${luaJIT_configure_options[@]}"
+    printf "\n"
+    make amalg "${luaJIT_configure_options[@]}" || exit 1
+    make install "${luaJIT_configure_options[@]}" || exit 1
 
     # luajit beta versions do not symlink luajit to the built binary,
     # so we'll do that manually here, for convenience and also so
     # luarocks can install.
     if [[ $luaJIT_version == *"beta"* ]]; then
-      ln -sf luajit-$luaJIT_version $ASDF_INSTALL_PATH/bin/luajit
+      ln -sf luajit-"$luaJIT_version" "$ASDF_INSTALL_PATH"/bin/luajit
     fi
 
     ##########################################################################
 
-    cd $(dirname $lua_rocks_source_path)
-    tar zxf $lua_rocks_source_path || exit 1
-    cd $(untar_lua_rocks_path $install_type $lua_rocks_version $tmp_download_dir)
+    cd "$(dirname "$lua_rocks_source_path")" || exit 1
+    tar zxf "$lua_rocks_source_path" || exit 1
+    cd "$(untar_lua_rocks_path "$install_type" "$lua_rocks_version" "$tmp_download_dir")" || exit
+
+    local lua_rocks_configure_options=()
+    while IFS='' read -r line; do
+      lua_rocks_configure_options+=("$line")
+    done < <(construct_lua_rocks_configure_options)
 
-    local lua_rocks_configure_options="$(construct_lua_rocks_configure_options)"
     # set in os_based_configure_options
     # we unset it here because echo-ing changes the return value of the function
     unset ASDF_PKG_MISSING
 
-    echo "Building luarocks with options: $lua_rocks_configure_options"
-    ./configure $lua_rocks_configure_options || exit 1
+    printf "Building luarocks with options:"
+    printf " %q" "${lua_rocks_configure_options[@]}"
+    printf "\n"
+    ./configure "${lua_rocks_configure_options[@]}" || exit 1
     make build || make || exit 1
     make install || exit 1
   )
 }
 
 construct_luaJIT_configure_options() {
-  if [ "$LUAJIT_CONFIGURE_OPTIONS" = "" ]; then
-    local configure_options="$(os_based_luaJIT_configure_options) PREFIX=$install_path"
-
-    if [ "$LUAJIT_EXTRA_CONFIGURE_OPTIONS" != "" ]; then
-      configure_options="$configure_options $LUAJIT_EXTRA_CONFIGURE_OPTIONS"
-    fi
+  local configure_options=()
+  if [[ "${LUAJIT_CONFIGURE_OPTIONS-}" == "" ]]; then
+    # we want typical IFS splitting here,
+    # and we've set -f earlier, so this is fine.
+    # shellcheck disable=SC2206
+    configure_options=(
+      PREFIX="$install_path"
+      ${LUAJIT_EXTRA_CONFIGURE_OPTIONS-}
```

Comment:
I make a `shellcheck` lint exception here, because I really do want `LUAJIT_EXTRA_CONFIGURE_OPTIONS` to be split apart in all the usual ways, _and_ I disabled the one big footgun here which is having file globbing enabled.